### ISSUE TITLE
Test failure

### ISSUE THREAD

**Issue by eclipseo** [jethrogb/rust-cexpr#8]
Rust 1.29 + cexpr = "0.3.0"

```
/usr/bin/cargo test -j2 --release --no-fail-fast

Running `/usr/bin/rustc --crate-name cexpr src/lib.rs --emit=dep-info,link -C opt-level=3 --test -C metadata=37d5fd183bd82eb0 -C extra-filename=-37d5fd183bd82eb0 --out-dir /builddir/build/BUILD/cexpr-0.3.0/target/release/deps -L dependency=/builddir/build/BUILD/cexpr-0.3.0/target/releasedeps --extern clang_sys=/builddir/build/BUILD/cexpr-0.3.0/target/release/deps/libclang_sys-6ff92d40c4d7f232.rlib --extern nom=/builddir/build/BUILD/cexpr-0.3.0/target/release/deps/libnom-03124c02db76c4ca.rlib -Copt-level=3 -Cdebuginfo=2 -Clink-arg=-Wl,-z,relro,-z,now -L /usr/lib64`
error[E0423]: expected function, found type alias `CXFile`
   --> tests/clang.rs:152:15
    |
152 |     let mut file=CXFile(ptr::null_mut());
    |                  ^^^^^^
    |
    = note: can't use a type alias as a constructor

error[E0609]: no field `0` on type `*mut libc::c_void`
   --> tests/clang.rs:156:11
    |
156 |         && file.0!=ptr::null_mut()
    |                 ^ help: `file` is a native pointer; try dereferencing it: `(*file).0`

error: aborting due to 2 previous errors

Some errors occurred: E0423, E0609.
```

**Comment by jethrogb** [comment]
Which version of LLVM do you have installed?

**Comment by eclipseo** [comment]
7.0.0

**Comment by jethrogb** [comment]
Can you test again with the following changes applied:
```patch
diff --git a/Cargo.toml b/Cargo.toml
index 4d00108..9031a03 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -12,4 +12,4 @@ keywords = ["C","expression","parser"]
 nom = {version = "^4", features = ["verbose-errors"] }
 
 [dev-dependencies]
-clang-sys = "0.11.0"
+clang-sys = "0.24.0"
diff --git a/tests/clang.rs b/tests/clang.rs
index 8578811..93cb8dc 100644
--- a/tests/clang.rs
+++ b/tests/clang.rs
@@ -149,11 +149,11 @@ unsafe fn visit_children<F>(cursor: CXCursor, mut f: F)
 
 unsafe fn location_in_scope(r: CXSourceRange) -> bool {
        let start=clang_getRangeStart(r);
-       let mut file=CXFile(ptr::null_mut());
+       let mut file: CXFile=ptr::null_mut();
        clang_getSpellingLocation(start,&mut file,ptr::null_mut(),ptr::null_mut(),ptr::null_mut());
        clang_Location_isFromMainFile(start)!=0
                && clang_Location_isInSystemHeader(start)==0
-               && file.0!=ptr::null_mut()
+               && file!=ptr::null_mut()
 }
 
 /// tokenize_range_adjust can be used to work around LLVM bug 9069
```

**Comment by eclipseo** [comment]
I was actually testing a very similar patch.

**Comment by eclipseo** [comment]
Success: https://copr-be.cloud.fedoraproject.org/results/eclipseo/rav1e/fedora-rawhide-x86_64/00804856-rust-cexpr/builder-live.log

Thanks for your reactivity!

**Comment by jethrogb** [comment]
Fixed in 0.3.1

**Comment by jethrogb** [comment]
Fixed in 0.3.1, sorry for the delay
